libraries/abstractions/common_io/test/test_iot_uart.c

```diff
@@ -0,0 +1,817 @@
+
+/*-----------------------------------------------------------*/
+/**
+ * This is Assisted Test function to test the asynchronous read/write with UART by doing a
+ * loopback with RPI3. The Tx and Rx pin on the vendor board are connected to RX and TX of
+ * RPI3. Test is run by transmitting some bytes on write async and checking
+ * if same characters are read asynchronously on RPI3. Validation happens on RPI3
+ * This test also checks if we are getting Tx bytes in flight.
+ */
+TEST( TEST_IOT_UART, AFQP_AssistedIotUARTWriteAsync )
+{
+    IotUARTHandle_t xUartHandle;
+    int32_t lIoctl, lWrite, lClose, lRead, lTransferAmount_1, lTransferAmount_2;
+    uint8_t i;
+    uint8_t cpBufferLarge[ testIotUART_BUFFER_LENGTH_LARGE + 1 ] = { 0 };
+
+    for (i = 0; i < testIotUART_BUFFER_LENGTH_LARGE - 1; i++)
+    {
+        cpBufferLarge[i] = 0xAA;
+    }
+    cpBufferLarge[testIotUART_BUFFER_LENGTH_LARGE ] = '\n';
+
+    xUartHandle = iot_uart_open( ustestIotUartPort );
+    TEST_ASSERT_NOT_EQUAL( NULL, xUartHandle );
+
+    if( TEST_PROTECT() )
+    {
+        lWrite = iot_uart_write_async( xUartHandle, cpBufferLarge, testIotUART_BUFFER_LENGTH_LARGE );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lWrite );
+
+        lIoctl = iot_uart_ioctl( xUartHandle, eGetTxNoOfbytes, &lTransferAmount_1 );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+        TEST_ASSERT_GREATER_THAN( 0, lTransferAmount_1  );
+
+        /* Delay for 1 sec. */
+        vTaskDelay(testIotUART_DELAY);
+
+        lIoctl = iot_uart_ioctl( xUartHandle, eGetTxNoOfbytes, &lTransferAmount_2 );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+        TEST_ASSERT_GREATER_THAN( lTransferAmount_1, lTransferAmount_2  );
+
+        /* Delay for 1 sec. */
+        vTaskDelay(testIotUART_DELAY);
+    }
+
+    lClose = iot_uart_close( xUartHandle );
+    TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lClose );
+}
+/*-----------------------------------------------------------*/
+
+/*-----------------------------------------------------------*/
+/**
+ * @brief Test function to test the asynchronous read/write with UART by doing a
+ * hardware loopback. The Tx and Rx pin on the vendor board are shorted with a
+ * connector and the test is run by transmitting some bytes on write sync and checking
+ * if same characters are read asynchronously.
+ * This test also checks if we are getting Rx bytes in flight.
+ */
+TEST( TEST_IOT_UART, AFQP_IotUARTWriteSyncReadAsyncLoopbackTest )
+{
+    IotUARTHandle_t xUartHandle;
+    int32_t lIoctl, lWrite, lClose, lRead, lTransferAmount;
+    BaseType_t xCallbackReturn;
+
+    uint8_t cpBuffer[ testIotUART_BUFFER_LENGTH + 1 ] = { 0 };
+    uint8_t cpBufferRead[ testIotUART_BUFFER_LENGTH + 1 ] = { 0 };
+    uint8_t uStringCompare = 1;
+
+    strncpy( (char*)cpBuffer, testIotUART_BUFFER_STRING, testIotUART_BUFFER_LENGTH );
+
+    xUartHandle = iot_uart_open( ustestIotUartPort );
+    TEST_ASSERT_NOT_EQUAL( NULL, xUartHandle );
+
```

Comment:
Could you keep Test protect? It was there in the original file.

Reply:
Same as above